cocotb/queue.py

```diff
@@ -138,6 +147,28 @@ def get_nowait(self) -> T:
         self._wakeup_next(self._putters)
         return item
 
+    async def peek(self) -> T:
+        """
+        Return a copy of the first item in the queue. If the queue is empty
+        wait until it is not.
+        """
+        while self.empty():
+            event = Event('{} get'.format(type(self).__name__))
+            self._getters.append((event, cocotb.scheduler._current_task))
+            await event.wait()
+        return self.peek_nowait()
+
+    def peek_nowait(self) -> T:
+        """
+        Return the first item in the queue. If the queue is empty
+        raise QueueEmpty()
+        """
+        if self.empty():
+            raise QueueEmpty()
+        item = self._peek()
+        self._wakeup_next(self._putters)
```

Comment:
Why do you wakeup the other events after a peek?

Reply:
Monkey see, monkey do. That shouldn't be in there.

Reply:
I've removed this code from a new pull request I"m entering